pyomo/contrib/gdpopt/tests/test_ldsda.py

```diff
@@ -0,0 +1,50 @@
+from pyomo.environ import SolverFactory, value, Var, Constraint, TransformationFactory
+from pyomo.gdp import Disjunct
+import pyomo.common.unittest as unittest
+from pyomo.contrib.gdpopt.tests.four_stage_dynamic_model import build_model
+
+
+class TestGDPoptLDSDA(unittest.TestCase):
+    """Real unit tests for GDPopt"""
+
+    @unittest.skipUnless(
+        SolverFactory('gams').available() and SolverFactory('gams').license_is_valid(),
+        "gams solver not available",
+    )
+    def test_solve_four_stage_dynamic_model(self):
+
+        model = build_model(mode_transfer=True)
+
+        # Discretize the model using dae.collocation
+        discretizer = TransformationFactory('dae.collocation')
+        discretizer.apply_to(model, nfe=10, ncp=3, scheme='LAGRANGE-RADAU')
```

Comment:
@AlbertLee125 could you try working around the DAE/GDP issue by moving the discretization transformation into the `build_model` function? If you discretize the model after the `ContinuousSet` and `DerivativeVar` components are declared but before the disjuncts then the differential equations in the disjuncts should contain the expanded set of discretization points.